src/node/blockstorage.h

```diff
@@ -5,15 +5,36 @@
 #ifndef BITCOIN_NODE_BLOCKSTORAGE_H
 #define BITCOIN_NODE_BLOCKSTORAGE_H
 
+#include <cstdint>
 #include <vector>
 
 #include <fs.h>
+#include <protocol.h> // For CMessageHeader::MessageStartChars
 
 class ArgsManager;
+class CBlock;
+class CBlockIndex;
+class CBlockUndo;
+class CChain;
+class CChainParams;
 class ChainstateManager;
+struct FlatFilePos;
+namespace Consensus {
+struct Params;
+}
 
 static constexpr bool DEFAULT_STOPAFTERBLOCKIMPORT{false};
 
+/** Functions for disk access for blocks */
+bool ReadBlockFromDisk(CBlock& block, const FlatFilePos& pos, const Consensus::Params& consensusParams);
+bool ReadBlockFromDisk(CBlock& block, const CBlockIndex* pindex, const Consensus::Params& consensusParams);
+bool ReadRawBlockFromDisk(std::vector<uint8_t>& block, const FlatFilePos& pos, const CMessageHeader::MessageStartChars& message_start);
```

Comment:
https://github.com/bitcoin/bitcoin/pull/21575/commits/fa0c7d9ad24d3c9515d3f9c136af4071cbd79055

It looks like this is unused outside of the `src/node/blockstorage.cpp` unit, so you could make it internal to that if you wanted to.

Reply:
I'll keep this one public for symmetry with the other read* functions